### ISSUE TITLE
jqwik hangs during generation of small Sets

### ISSUE THREAD

**Issue by michailgames** [jlink/jqwik#65]
## Testing Problem

The following Set provider:

```
@Provide
Arbitrary<Set<SomeEnum>> smallSet() {
    var allowed = List.of(SomeEnum.A, SomeEnum.B, SomeEnum.C);
    return Arbitraries.of(allowed).set();
}
```

seems to hang forever when trying to generate the sets.
But, when ` Arbitraries.of(allowed).set().ofMaxSize(3)` is used, everything is fine.

## Suggested Solution

Probably max set size should be set automatically when generating set from constant number of elements.

**Comment by jlink** [comment]
Thanks for the catch!
The list jqwik should do is not trying to generate values endlessly. 

I'll have to figure out if the automatic setting of maxSize is possible (in a generic way).

**Comment by jlink** [comment]
Fixed.

Available in 1.2.0-SNAPSHOT
